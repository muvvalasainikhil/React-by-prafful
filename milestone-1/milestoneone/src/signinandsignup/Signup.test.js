import React from "react";
import { MemoryRouter as Router, withRouter } from "react-router-dom";
import { mount } from "enzyme";
import { shallow } from "enzyme";
import { configure } from "enzyme";
import Signup from "../signinandsignup/Signup";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

it("Render", () => {
  const wrapper = shallow(<Signup />);
  const h = wrapper.find("h1");
  const result = h.text();
  expect(result).toBe("Register");
});
it("Submit Fucntion", () => {
  const onSubmitFun = jest.fn();
  const wrapper = mount(<form onSubmit={onSubmitFun}></form>);
  const form = wrapper.find("form");
  form.simulate("submit");
  expect(onSubmitFun).toHaveBeenCalledTimes(1);
});
it("username field", () => {
  const wrapper = mount(<input type="text" name="username" />);
  const input = wrapper.find("input");
  expect(input).toHaveLength(1);
  expect(input.prop("type")).toEqual("text");
  expect(input.prop("name")).toEqual("username");
});
it("Password field", () => {
  const wrapper = mount(<input type="password" name="password" />);
  const input = wrapper.find("input");
  expect(input).toHaveLength(1);
  expect(input.prop("type")).toEqual("password");
  expect(input.prop("name")).toEqual("password");
});
it("Confirm Password field", () => {
  const wrapper = mount(<input type="password" name="confirm_password" />);
  const input = wrapper.find("input");
  expect(input).toHaveLength(1);
  expect(input.prop("type")).toEqual("password");
  expect(input.prop("name")).toEqual("confirm_password");
});
it("Email field", () => {
  const wrapper = mount(<input type="email" name="email" />);
  const input = wrapper.find("input");
  expect(input).toHaveLength(1);
  expect(input.prop("type")).toEqual("email");
  expect(input.prop("name")).toEqual("email");
});
it("Button field", () => {
  const wrapper = mount(<button type="submit">Login</button>);
  const input = wrapper.find("button");
  const result = input.text();
  expect(result).toBe("Login");
  expect(input.prop("type")).toEqual("submit");
});
// test("mathes Snapshot", () => {
//   const tree = renderer
//     .create(
//       <Router>
//         <Signup />
//       </Router>
//     )
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });
