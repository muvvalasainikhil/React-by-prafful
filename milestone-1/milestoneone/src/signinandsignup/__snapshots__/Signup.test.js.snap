// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes Snapshot 1`] = `
<div
  className="center"
>
  <h1>
    SIGNUP
  </h1>
  <form
    onSubmit={[Function]}
  >
    <table>
      <thead />
      <tbody>
        <tr>
          <td>
            <label>
              <b>
                Username
              </b>
            </label>
          </td>
          <td>
            <input
              name="username"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value=""
            />
            <span
              className="error"
            >
              
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <label>
              <b>
                Email
              </b>
            </label>
          </td>
          <td>
            <input
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              type="email"
              value=""
            />
            <span
              className="error"
            >
              
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <label>
              <b>
                Password
              </b>
            </label>
          </td>
          <td>
            <input
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
              value=""
            />
            <span
              className="error"
            >
              
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <label>
              <b>
                Confirm Password
              </b>
            </label>
          </td>
          <td>
            <input
              name="confirm_password"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
              value=""
            />
            <span
              className="error"
            >
              
            </span>
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/"
              onClick={[Function]}
            >
              Already User?
            </a>
          </td>
          <td>
            <button>
              Register
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;
